gateway-protocol-impl/src/test/java/io/camunda/zeebe/gateway/protocol/ServiceConfigTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.gateway.protocol;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonCreator.Mode;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.grpc.MethodDescriptor;
+import io.grpc.ServiceDescriptor;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import org.assertj.core.api.Condition;
+import org.assertj.core.condition.VerboseCondition;
+import org.junit.jupiter.api.Test;
+
+final class ServiceConfigTest {
+
+  @Test
+  void shouldHaveAPolicyForAllServiceMethods() throws IOException {
+    // given
+    final ObjectMapper objectMapper = new ObjectMapper();
+    final ServiceDescriptor serviceDescriptor = GatewayGrpc.getServiceDescriptor();
+    final Collection<MethodDescriptor<?, ?>> methods = serviceDescriptor.getMethods();
+    final PartialServiceConfig serviceConfig =
+        objectMapper.readValue(
+            ClassLoader.getSystemClassLoader().getResource("gateway-service-config.json"),
+            PartialServiceConfig.class);
+
+    // when - then
+    for (final MethodDescriptor<?, ?> method : methods) {
+      assertThat(serviceConfig).has(hasRetryPolicyFor(method));
+    }
+  }
+
+  private Condition<? super PartialServiceConfig> hasRetryPolicyFor(
+      final MethodDescriptor<?, ?> method) {
+    final MethodName expected = new MethodName(method.getServiceName(), method.getBareMethodName());
+    return VerboseCondition.verboseCondition(
+        config ->
+            config.methodConfig.stream()
+                .filter(m -> m.name.contains(expected))
+                .anyMatch(m -> m.retryPolicy != null),
+        String.format(
+            "a service config with a retry policy for method '%s'", method.getFullMethodName()),
+        config -> " but no such retry policy was found");
```

Comment:
👍 I like the test

🔧 While you and I understand the need for this now, it might get lost over time leading to confusion when this test fails. Could we add some additional context to this message describing what this is or why it's needed that all methods have a retry policy?

🔧 It would also be very helpful to describe how to resolve this, so where should this policy be defined.

Reply:
New error message:

```
Expecting actual:
  PartialServiceConfig{methodConfig=[PartialMethodConfig{name=[MethodName{service='gateway_protocol.Gateway', method='ActivateJobs'}, MethodName{service='gateway_protocol.Gateway', method='CancelProcessInstance'}, MethodName{service='gateway_protocol.Gateway', method='CompleteJob'}, MethodName{service='gateway_protocol.Gateway', method='DeleteResource'}, MethodName{service='gateway_protocol.Gateway', method='EvaluateDecision'}, MethodName{service='gateway_protocol.Gateway', method='FailJob'}, MethodName{service='gateway_protocol.Gateway', method='ResolveIncident'}, MethodName{service='gateway_protocol.Gateway', method='SetVariables'}, MethodName{service='gateway_protocol.Gateway', method='StreamActivatedJobs'}, MethodName{service='gateway_protocol.Gateway', method='Topology'}], retryPolicy={maxAttempts=5.0, initialBackoff=0.1s, maxBackoff=5s, backoffMultiplier=3.0, retryableStatusCodes=[UNAVAILABLE, RESOURCE_EXHAUSTED, DEADLINE_EXCEEDED]}}, PartialMethodConfig{name=[MethodName{service='gateway_protocol.Gateway', method='BroadcastSignal'}, MethodName{service='gateway_protocol.Gateway', method='CreateProcessInstance'}, MethodName{service='gateway_protocol.Gateway', method='CreateProcessInstanceWithResult'}, MethodName{service='gateway_protocol.Gateway', method='DeployProcess'}, MethodName{service='gateway_protocol.Gateway', method='DeployResource'}, MethodName{service='gateway_protocol.Gateway', method='ModifyProcessInstance'}, MethodName{service='gateway_protocol.Gateway', method='PublishMessage'}, MethodName{service='gateway_protocol.Gateway', method='ThrowError'}, MethodName{service='gateway_protocol.Gateway', method='UpdateJobRetries'}, MethodName{service='gateway_protocol.Gateway', method='UpdateJobTimeout'}], retryPolicy={maxAttempts=5.0, initialBackoff=0.1s, maxBackoff=5s, backoffMultiplier=3.0, retryableStatusCodes=[UNAVAILABLE, RESOURCE_EXHAUSTED]}}]}
to have a service config with a retry policy for method 'gateway_protocol.Gateway/ActivateJobs' defined but no such retry policy was found in 'file:/home/nicolas/src/github.com/camunda-cloud/zeebe/gateway-protocol-impl/target/classes/gateway-service-config.json'
```

(obviously the path will change based on your local machine ;))